test/ERC20RootVault.test.ts

```diff
@@ -0,0 +1,1149 @@
+                            .connect(this.deployer)
+                            .approve(this.subject.address, BigNumber.from(100));
+                        await expect(
+                            this.subject
+                                .connect(this.deployer)
+                                .deposit(
+                                    [BigNumber.from(1), BigNumber.from(1)],
+                                    BigNumber.from(2),
+                                    []
+                                )
+                        ).to.be.revertedWith(Exceptions.LIMIT_UNDERFLOW);
+                    });
+                });
+
+                describe("when lpAmount is 0", () => {
+                    it(`reverts with ${Exceptions.VALUE_ZERO}`, async () => {
+                        await this.subject
+                            .connect(this.admin)
+                            .addDepositorsToAllowlist([this.deployer.address]);
+                        await this.weth
+                            .connect(this.deployer)
+                            .approve(
+                                this.subject.address,
+                                BigNumber.from(10001)
+                            );
+                        await this.usdc
+                            .connect(this.deployer)
+                            .approve(
+                                this.subject.address,
+                                BigNumber.from(10001)
+                            );
+                        await expect(
+                            this.subject
+                                .connect(this.deployer)
+                                .deposit(
+                                    [
+                                        BigNumber.from(10001),
+                                        BigNumber.from(10001),
+                                    ],
+                                    BigNumber.from(0),
+                                    []
+                                )
+                        ).not.to.be.reverted;
+
+                        await this.subject
+                            .connect(this.admin)
+                            .addDepositorsToAllowlist([this.deployer.address]);
+                        await this.weth
+                            .connect(this.deployer)
+                            .approve(
+                                this.subject.address,
+                                BigNumber.from(10001)
+                            );
+                        await this.usdc
+                            .connect(this.deployer)
+                            .approve(
+                                this.subject.address,
+                                BigNumber.from(10001)
+                            );
+                        await expect(
+                            this.subject
+                                .connect(this.deployer)
+                                .deposit(
```

Comment:
why do we receive 0 lpTokens amount in the second deposit? We should have received 10001 lpTokens if I am right, because we make a non-zero deposit => it is strange that we get 0 lpTokens